ratis-proto/src/main/proto/Raft.proto

```diff
@@ -327,6 +327,7 @@ message DataStreamPacketHeaderProto {
   Type type = 3;
   repeated Option options = 4;
   uint64 dataLength = 5;
+  bytes clientId = 6;
```

Comment:
Since Streaming is not yet released, how about we take this chance to renumber the proto?  
```
  bytes clientId = 1;
  Type type = 2;
  uint64 streamId = 3;
  uint64 streamOffset = 4;
  uint64 dataLength = 5;
  repeated Option options = 6;
```
Then, the proto and the java class will have the same order
```
public DataStreamRequestHeader(ClientId clientId, Type type, long streamId, long streamOffset, long dataLength,
       WriteOption... options) {
```